### PR TITLE
Add join without volatile qualifier in BoundingBoxReduce

### PR DESCRIPTION
**Pull request by streeve**
Disabled in Kokkos develop

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
Do you want to support kokkos version less than 3.7?

**Comment by streeve** [comment]
@abisner can you review?

File: src/Picasso_FacetGeometry.hpp [hunk 1]
```diff
@@ -147,6 +147,18 @@ class FacetGeometry
                             fmax( facets( f, 1, 2 ), facets( f, 2, 2 ) ) ) );
         }
 
+#if KOKKOS_VERSION >= 30700
```
Comment (dalg24) [inline src/Picasso_FacetGeometry.hpp:150]: No that is the opposite, you want that overload always and guard the one with volatile-qualified argument with `#if KOKKOS_VERSION < 30700`

Comment (streeve) [reply src/Picasso_FacetGeometry.hpp]: Thanks - should be fixed

**Review by dalg24 — APPROVED** [review]

**Comment by streeve** [comment]
HIP failure unrelated

**Pull request merged**